### PR TITLE
feat: Showing deployment status card for helm type Devtron apps

### PR DESCRIPTION
**Pull request by sohel-rp**
# Description

These changes will enable deployment status card for helm type Devtron apps

Fixes - [AB#1845](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/1845)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All tests have been performed manually by accessing Devtron apps of type helm.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/details/appDetails/SourceInfo.tsx [hunk 1]
```diff
@@ -161,63 +168,67 @@ export function SourceInfo({
...
+                            </div>
+                            <div className="flex left column mw-140">
+                                <div className="fs-12 fw-4 cn-9">Deployment triggered</div>
+                                <div className="flexbox">
+                                    <span className="fs-13 mr-5 fw-6 cn-9">
+                                        {deploymentTriggerTime
+                                            ? moment(deploymentTriggerTime, 'YYYY-MM-DDTHH:mm:ssZ').fromNow()
+                                            : '-'}
+                                    </span>
+                                    {deploymentStatus === 'inprogress' && <Timer className="icon-dim-16 mt-4" />}
```
Comment (vivek-devtron) [inline src/components/app/details/appDetails/SourceInfo.tsx:225]: can we use the constant here

File: src/components/app/details/appDetails/SourceInfo.tsx [hunk 2]
```diff
@@ -161,63 +168,67 @@ export function SourceInfo({
...
+                                        placement="top"
+                                        content="Status of last triggered deployment"
+                                    >
+                                        <Question className="icon-dim-16 mt-2" />
+                                    </Tippy>
+                                </div>
+                                <div className="flexbox">
+                                    <span
+                                        className={`app-summary__status-name fs-14 mr-8 fw-6 f-${deploymentStatus} ${
+                                            deploymentStatus === 'inprogress' ? 'dc__loading-dots' : ''
```
Comment (vivek-devtron) [inline src/components/app/details/appDetails/SourceInfo.tsx:204]: can we use the constant here?

**Review by vivek-devtron — APPROVED** [review]

File: src/components/app/details/appDetails/appDetails.type.ts [hunk 3]
```diff
@@ -71,6 +71,7 @@ export interface DeploymentStatusDetailsType {
     statusFetchCount: number
     statusLastFetchedAt: string
     timelines: DeploymentStatusDetailsTimelineType[]
+    wfrStatus?: string
```
Comment (shivani170) [inline src/components/app/details/appDetails/appDetails.type.ts:74]: Can we rename it to more explanatory?

File: src/components/app/details/appDetails/AppDetails.tsx [hunk 4]
```diff
@@ -253,7 +250,25 @@ export const Details: React.FC<{
...
-                        processDeploymentStatusData(deploymentStatusDetailRes.result)
+
+                        if (deploymentStatusDetailRes.result) {
+                            if (response.result?.deploymentAppType === DeploymentAppType.helm) {
+                                setDeploymentStatusDetailsBreakdownData({
+                                    ...deploymentStatusDetailsBreakdownData,
+                                    deploymentStatus:
+                                        DEPLOYMENT_STATUS[deploymentStatusDetailRes.result.wfrStatus?.toUpperCase()],
+                                    deploymentStatusText:
+                                        deploymentStatusDetailRes.result.wfrStatus === 'Progressing'
```
Comment (shivani170) [inline src/components/app/details/appDetails/AppDetails.tsx:261]: Can we take this form constant if possible

File: src/components/app/details/appDetails/appDetails.type.ts (hunk 3 above)
Comment (sohel-rp) [reply src/components/app/details/appDetails/appDetails.type.ts:74]: This is coming from BE.

File: src/components/app/details/appDetails/SourceInfo.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/app/details/appDetails/SourceInfo.tsx]: done

**Review by shivani170 — APPROVED** [review]

**Pull request merged**